app/models/course_data/assignment.rb

```diff
@@ -70,4 +72,18 @@ def set_defaults_and_normalize
     return if article_title.nil?
     self.article_title = ArticleUtils.format_article_title(article_title, wiki)
   end
+
+  def set_sandbox_url
+    return unless wiki && user
+    # If the sandbox already exists, use that URL instead
+    existing = Assignment.where(course: course, article: article,
+                                wiki: wiki)
+                         .where.not(user: user).first
```

Comment:
This won't match for cases where the article doesn't exist, so the existing assignment has an `article_title` that is the same instead.